packages/zowe-explorer/src/Profiles.ts

```diff
@@ -1079,11 +1079,25 @@ export class Profiles extends ProfilesCache {
                 cmdArgs[prop] = serviceProfile.profile[prop] ? serviceProfile.profile[prop] : baseProfile.profile[prop];
             }
             if (baseProfile) {
-                cmdArgs.tokenType = serviceProfile.profile.tokenType ? serviceProfile.profile.tokenType: baseProfile.profile.tokenType;
-                cmdArgs.tokenValue = serviceProfile.profile.tokenValue ? serviceProfile.profile.tokenValue: baseProfile.profile.tokenValue;
+                cmdArgs.tokenType = serviceProfile.profile.tokenType
+                    ? serviceProfile.profile.tokenType
+                    : baseProfile.profile.tokenType;
+                cmdArgs.tokenValue = serviceProfile.profile.tokenValue
+                    ? serviceProfile.profile.tokenValue
+                    : baseProfile.profile.tokenValue;
             }
             if (commonApi.getSessionFromCommandArgument) {
-                session = await commonApi.getSessionFromCommandArgument(cmdArgs);
+                if (cmdArgs.tokenType === undefined || cmdArgs.tokenValue === undefined) {
+                    vscode.window.showInformationMessage(
+                        "Base Profile " +
+                            baseProfile.name +
+                            " is currently logged out. Please login and reload Zowe Explorer to be able to use it."
+                    );
+                    session = baseProfile.profile;
```

Comment:
same here with only looking to base profile, there should be a check as to where the token value came from that that profile should be the session

Reply:
So I tested this. I move the base profile because I needed the info if there is a base profile. If ever, they have the info in the service profile then it will be moved in line 1110.